packages/typography/src/Typography.tsx

```diff
@@ -117,15 +100,19 @@ Subtitle.displayName = 'Subtitle';
 /**
  * Body
  */
-type BodyProps = HTMLElementProps<'div'> & {
+type BodyProps = HTMLElementProps<'p'> & {
   /**
    * font-weight applied to typography element
    * default: `regular`
    */
   weight?: 'regular' | 'medium';
 };
 
-function Body({ children, className, weight = 'regular', ...rest }: BodyProps) {
+const Body: ExtendableBox<BodyProps, 'p'> = ({
+  className,
+  weight = 'regular',
+  ...rest
+}: BodyProps) => {
```

Comment:
We should add an `as` prop here. My understanding of the ticket is that we want to enable users to write, e.g. `<Body as='"span">` or `<Body as="p">` etc.

Reply:
Makes sense! Should we restrict this to just a few values that make sense for `Body`? Allowing `<Body as="h1" />` feels weird

Reply:
I feel like that'll get pretty hairy in Typescript. And we're pretty likely to miss one or two tags that do make sense. If someone wants a semantic `h1`  that looks like body text, then they can go ahead 😆

Reply:
What matters in this case is that developers can use any semantic tag regardless of the actual appearance on the page to create a clear page hierarchy for screen readers. Using an `h1` like this may actually be appropriate depending on the circumstance.